packages/blocks/src/api/utils.js

```diff
@@ -33,31 +33,38 @@ const ICON_COLORS = [ '#191e23', '#f8f9f9' ];
  * Determines whether the block's attributes are equal to the default attributes
  * which means the block is unmodified.
  *
- * @param {WPBlock} block Block Object
+ * @param {WPBlock}   block         Block Object
+ * @param {?string[]} attributeKeys The optional keys of the attributes to check.
  *
  * @return {boolean} Whether the block is an unmodified block.
  */
-export function isUnmodifiedBlock( block ) {
-	return Object.entries( getBlockType( block.name )?.attributes ?? {} ).every(
-		( [ key, definition ] ) => {
-			const value = block.attributes[ key ];
-
-			// Every attribute that has a default must match the default.
-			if ( definition.hasOwnProperty( 'default' ) ) {
-				return value === definition.default;
-			}
+export function isUnmodifiedBlock( block, attributeKeys ) {
```

Comment:
Might it better to leave `isUnmodifiedBlock` intact and create a new util with our implementation of whether a block can be considered empty?

Additionally, should the empty block function just check for

```
if ( definition.type === 'rich-text' ) {
	return ! value?.length;
}
```

Do you see any drawbacks on that?

Reply:
Third-party blocks could have content attributes that are not rich-text, (but would it able to call `onMerge`?). I think it's better to keep it a little bit generic for now. 😅

Reply:
I think many third party blocks are also unlikely to have `__experimentalRole` defined because of the way it's experimental.

Reply:
Just FYI it's been stablised now but Dan's point still stands:

https://github.com/WordPress/gutenberg/pull/65484

Reply:
Yeah, I'm just trying to be safe and future-compatible here. (On the other hand, an attribute could also have the `rich-text` type but not be considered content.)

Thanks for the link! I can rebase this to use the stabilized function.